compiler/rustc_session/src/filesearch.rs

```diff
@@ -78,10 +78,16 @@ fn current_dll_path() -> Result<PathBuf, String> {
                 if libc::dladdr(addr, &mut info) == 0 {
                     return Err("dladdr failed".into());
                 }
-                if info.dli_fname.is_null() {
-                    return Err("dladdr returned null pointer".into());
-                }
-                let bytes = CStr::from_ptr(info.dli_fname).to_bytes();
+                #[cfg(target_os = "cygwin")]
+                let fname_ptr = info.dli_fname.as_ptr();
+                #[cfg(not(target_os = "cygwin"))]
+                let fname_ptr = {
+                    if info.dli_fname.is_null() {
```

Comment:
Hm, this can be null? I can't find that in the docs. Miri seems to assume it's never null...

(This is pre-existing, not new in this PR.)

Reply:
> If the address specified in addr could not be matched to a shared
       object, then these functions return 0.  In this case, an error
       message is not available via [dlerror(3)](https://man7.org/linux/man-pages/man3/dlerror.3.html).

My read of that is that either it finds a shared object, in which case dli_fname is valid, or else it will return 0 (but not tell you why in dlerror).

Reply:
`dladdr` can return null, yes. That's handled above. I am talking about `dli_fname` being null, which is different and which should not be possible according to the docs.

Reply:
I opened a PR for this: https://github.com/rust-lang/rust/pull/141239